coding/python_testing.rst

```diff
@@ -4,6 +4,156 @@ Python Unit Testing
 
 This page provides technical guidance to developers writing unit tests for DM's Python code base.
 See :doc:`unit_test_policy` for an overview of LSST Stack testing.
+If you have legacy DM :mod:`unittest` ``suite``-based code please refer to tech note `SQR-012`_ for porting instructions.
+LSST tests should be written using the :mod:`unittest` framework, with default test discovery, and should support being run using the `pytest`_ test runner.
+
+.. _SQR-012: http://sqr-012.lsst.io
+.. _pytest: http://pytest.org
+
+Introduction to ``unittest``
+============================
+
+This document will not attempt to explain full details of how to use :mod:`unittest` but instead shows common scenarios encountered in the LSST codebase.
+
+A simple :mod:`unittest` example is shown below:
+
+.. literalinclude:: unit_test_snippets/unittest_basic_example.py
+   :language: python
+
+The important things to note in this example are:
+
+* If the test is being executed using ``python`` from the command line the :py:func:`unittest.main` call performs the test discovery and executes the tests, setting exit status to non-zero if any of the tests fail.
+* Test classes are executed in the order in which they appear in the test file.
+  In this case the tests in :class:`DemoTestCase1` will be executed before those in :class:`DemoTestCase2`.
+* Test classes must, ultimately, inherit from :class:`unittest.TestCase` in order to be discovered, and tests themselves must be methods that start with ``test``.
```

Comment:
Ultimately, yes, but LSST recommends inheriting from lsst.utils.tests.TestCase so could you please list that first? That way we train folks right away.